hydra-node/test/Hydra/Chain/DirectSpec.hs

```diff
@@ -3,58 +3,68 @@
+-- we need either to turn those into proper Unit tests, eg. not requiring a full-blown mock server to run,
+-- or drop them altogether in favor of `DirectChainSpec` tests which test the real thing.
 module Hydra.Chain.DirectSpec where
 
 import Hydra.Prelude
 import Test.Hydra.Prelude
 
+import Cardano.Ledger.Alonzo.Tx (ValidatedTx)
 import Control.Concurrent (newEmptyMVar, putMVar, takeMVar)
-import Control.Exception (ErrorCall)
 import Hydra.Chain (
   Chain (..),
   HeadParameters (HeadParameters),
+  InvalidTxError (NoSeedInput),
   OnChainTx (OnAbortTx, OnInitTx),
   PostChainTx (AbortTx, InitTx),
  )
-import Hydra.Chain.Direct (withDirectChain)
+import Hydra.Chain.Direct (DirectChainLog, withDirectChain)
 import Hydra.Chain.Direct.MockServer (withMockServer)
-import Hydra.Chain.Direct.Util (retrying)
+import Hydra.Chain.Direct.Util (Era, VerificationKey, retry)
 import Hydra.Chain.Direct.Wallet (generateKeyPair)
 import Hydra.Chain.Direct.WalletSpec (genPaymentTo)
-import Hydra.Logging (nullTracer)
+import Hydra.Ledger.Cardano (CardanoTx, NetworkMagic)
+import Hydra.Logging (showLogsOnFailure)
 import Hydra.Party (Party, deriveParty, generateKey)
 import Test.QuickCheck (generate)
 
 spec :: Spec
 spec = do
-  it "can init and abort a head given nothing has been committed" $ do
-    calledBackAlice <- newEmptyMVar
-    calledBackBob <- newEmptyMVar
-    aliceKeys@(aliceVk, _) <- generateKeyPair
-    bobKeys@(bobVk, _) <- generateKeyPair
-    withMockServer $ \magic iocp socket submitTx -> do
-      let cardanoKeys = [] -- TODO(SN): this should matter
-      withDirectChain nullTracer magic iocp socket aliceKeys alice cardanoKeys (putMVar calledBackAlice) $ \Chain{postTx} -> do
-        withDirectChain nullTracer magic iocp socket bobKeys bob cardanoKeys (putMVar calledBackBob) $ \_ -> do
-          let parameters = HeadParameters 100 [alice, bob, carol]
-          generate (genPaymentTo magic aliceVk) >>= submitTx
-          generate (genPaymentTo magic bobVk) >>= submitTx
-
-          failAfter 5 $
-            retrying @ErrorCall $ postTx $ InitTx parameters
-          failAfter 5 $
-            takeMVar calledBackAlice `shouldReturn` OnInitTx 100 [alice, bob, carol]
-          failAfter 5 $
-            takeMVar calledBackBob `shouldReturn` OnInitTx 100 [alice, bob, carol]
-
-          postTx $ AbortTx mempty
-
-          failAfter 5 $
-            takeMVar calledBackAlice `shouldReturn` OnAbortTx
-          failAfter 5 $
-            takeMVar calledBackBob `shouldReturn` OnAbortTx
+  it "can init and abort a head given nothing has been committed" $
+    failAfter 10 $
+      showLogsOnFailure $ \tracer -> do
+        calledBackAlice <- newEmptyMVar
+        calledBackBob <- newEmptyMVar
+        aliceKeys@(aliceVk, _) <- generateKeyPair
+        bobKeys <- generateKeyPair
+        withMockServer $ \magic iocp socket submitTx -> do
+          let cardanoKeys = [] -- TODO(SN): this should matter
```

Comment:
Not too sure about this anymore. i.e. I would not know what for we would need the cardano verification keys in this scenario. In the real world these are also only used for distributing initial funds.. right?

Reply:
No, they are used for generating the `InitTx`. I say this does not matter in this particular testing scenario, whose scope and usefulness is not clear anyway.